src/Esprima/Scanner.cs

```diff
@@ -1084,6 +1051,27 @@ public bool IsImplicitOctalLiteral()
             return true;
         }
 
+        private string ScanLiteralPart(Func<char, bool> check)
+        {
+            string num = "";

```

Comment:
We should prevent all these strings allocations by passing the outer StringBuilder.

Reply:
fixed

Reply:
For more perfomance, we could parse direcly into an int/long/double etc, without using a string. But I think that can be done in a another task